common/src/api/internal/mod.rs

```diff
@@ -5,3 +5,93 @@
 //! Internally facing APIs.
 
 pub mod nexus;
+
+use futures::future::ready;
+use futures::stream::StreamExt;
+use schemars::JsonSchema;
+use serde::Serialize;
+use uuid::Uuid;
+
+use super::external::ObjectStream;
+
+pub async fn to_list<T, U>(object_stream: ObjectStream<T>) -> Vec<U>
+where
+    T: Into<U>,
+{
+    object_stream
+        .filter(|maybe_object| ready(maybe_object.is_ok()))
+        .map(|maybe_object| maybe_object.unwrap().into())
+        .collect::<Vec<U>>()
+        .await
+}
+
+// Sagas
+//
+// These are currently only intended for observability by developers.  We will
+// eventually want to flesh this out into something more observable for end
+// users.
```

Comment:
Nit: This could be a rustdoc comment - we're probably okay documenting this for internal usage, since it's not part of the external API anymore.